### ISSUE TITLE
App freezes when more than 20 chars entered in textfield in secure entry mode.

### ISSUE THREAD

**Issue by azizuysal** [CosmicMind/Material#623]

**Comment by danieldahan** [comment]
Hey, thank you for this. What version of Material are you using? And by `App`, do you mean the sample app project, or your own?

**Comment by azizuysal** [comment]
It doesn't happen in the sample app. But in my own app (apps actually, happens in more than one app), secure text entry password field freezes when more characters than visible on screen entered. Instead of scrolling the chars horizontally, like the regular textfield does, it freezes. When I pause Xcode debugging and look at frames, I see that it's frozen on line 374 of TextField.swift, at super.layourSubviews() call (screenshot attached). I have two TextFields in a storyboard file, marked up as TextField/Material in IB, and connected to IBOutlets in my controller. I'm using Xcode 8.1, iOS SDK 10, and using AutoLayout. In the code, I add/change a few things, fonts and colors, but that's all. I just upgraded to latest 2.4.0, but same result.

<img width="824" alt="screen shot 2016-12-05 at 10 56 58 am" src="https://cloud.githubusercontent.com/assets/1683443/20898042/a4c7b9ee-bad9-11e6-8ff9-e1fd0e6d8671.png">

**Comment by danieldahan** [comment]
Can you send me a sample then. I will need to reproduce this, and if it is not happening in the sample, I need to see what is going on in your end. I'd be happy to help :)

**Comment by danieldahan** [comment]
Hey I haven't heard from you. If you still have the issue, please reopen or create a new issue. Thank you :)

**Comment by azizuysal** [comment]
Sorry I was busy with work. I did some tests and I can easily reproduce this. Just set font size to anything above 17 and set secure text entry on, and it will happen every single time when you enter more chars than visible and the field tries to scroll horizontally. I tested with both UIFont.systemFont(ofSize: 20) and RobotoFont.regular(with: 20) with identical results in a test project. If you cannot reproduce, I can send you the test project.
